### ISSUE TITLE
Incorrect documentation

### ISSUE THREAD

**Issue by azhang** [prometheus-operator/prometheus-operator#3585]
> While ServiceMonitors must live in the same namespace as the Prometheus resource, discovered targets may come from any namespace. This allows cross-namespace monitoring use cases, for example, for meta-monitoring. Use the namespaceSelector of the ServiceMonitorSpec to restrict the namespaces from which Endpoints objects may be discovered.

https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/user-guides/running-exporters.md

This isn't a limitation if the following is set.

```
        prometheusSpec:
          serviceMonitorSelectorNilUsesHelmValues: false 
```

**Comment by simonpasquier** [comment]
Can you elaborate a bit? What do you think is wrong in the quoted text?

**Comment by azhang** [comment]
ServiceMonitors do not have to live in the same namespace as the Prometheus resource. I deploy ServiceMonitors in the same namespace as the targets (eg "prod", "staging"), which is different than the namespace in which my Prometheus resource is defined (eg "monitoring") and it works perfectly if I set `serviceMonitorSelectorNilUsesHelmValues: false`. This particular setting is mentioned in some other github issues – I recall one of them mentioning that bitnami/prometheus-operator sets it `false` by default because that's the more common usage.

**Comment by simonpasquier** [comment]
Right, I see how the sentence can be confusing. PR welcome to clarify this!

**Comment by nrchakradhar** [comment]
At this link https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/user-guides/getting-started.md#related-resources
its mentioned that 

> The Prometheus resource includes a field called serviceMonitorSelector, which defines a selection of ServiceMonitors to be used. By default and before the version v0.19.0, ServiceMonitors must be installed in the same namespace as the Prometheus instance. **With the Prometheus Operator v0.19.0 and above, ServiceMonitors can be selected outside the Prometheus namespace** via the serviceMonitorNamespaceSelector field of the Prometheus resource.

While in the https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/user-guides/running-exporters.md the mentioned portion is not updated.

I can update the same to be aligned with [getting-started](https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/user-guides/getting-started.md#related-resources)

**serviceMonitorSelectorNilUsesHelmValues** is not part of Prometheus Spec, but part of Helm chart values.yaml. When set to false, the prometheus spec will have `serviceMonitorSelector: {} `set to nil
@azhang @simonpasquier  if this is fine I can raise a Documentation PR.

**Comment by simonpasquier** [comment]
@nrchakradhar sounds good to me. I wouldn't mention `serviceMonitorSelectorNilUsesHelmValues` since it's only relevant for Helm users.
